src/components/RoomNameInput/index.js

```diff
@@ -1,68 +1,40 @@
-import React, {useState} from 'react';
-import _ from 'underscore';
+import React from 'react';
 import CONST from '../../CONST';
 import TextInput from '../TextInput';
 import useLocalize from '../../hooks/useLocalize';
 import * as roomNameInputPropTypes from './roomNameInputPropTypes';
+import InputWrapper from '../Form/InputWrapper';
+import getOperatingSystem from '../../libs/getOperatingSystem';
 import * as RoomNameInputUtils from '../../libs/RoomNameInputUtils';
 
-function RoomNameInput({isFocused, autoFocus, disabled, errorText, forwardedRef, value, onBlur, onChangeText, onInputChange, shouldDelayFocus}) {
+function RoomNameInput({isFocused, autoFocus, disabled, errorText, forwardedRef, onBlur, shouldDelayFocus, inputID}) {
     const {translate} = useLocalize();
 
-    const [selection, setSelection] = useState();
+    const keyboardType = getOperatingSystem() === CONST.OS.IOS ? CONST.KEYBOARD_TYPE.ASCII_CAPABLE : CONST.KEYBOARD_TYPE.VISIBLE_PASSWORD;
 
-    /**
-     * Calls the onChangeText callback with a modified room name
-     * @param {Event} event
-     */
-    const setModifiedRoomName = (event) => {
-        const roomName = event.nativeEvent.text;
-        const modifiedRoomName = RoomNameInputUtils.modifyRoomName(roomName);
-        onChangeText(modifiedRoomName);
-
-        // if custom component has onInputChange, use it to trigger changes (Form input)
-        if (_.isFunction(onInputChange)) {
-            onInputChange(modifiedRoomName);
-        }
-
-        // Prevent cursor jump behaviour:
-        // Check if newRoomNameWithHash is the same as modifiedRoomName
-        // If it is then the room name is valid (does not contain unallowed characters); no action required
-        // If not then the room name contains unvalid characters and we must adjust the cursor position manually
-        // Read more: https://github.com/Expensify/App/issues/12741
```

Comment:
This is no longer an issue with the new solution?

Reply:
Yes, with current approach if we are parsing the value of the input, it's completely separated from the value we display, so it's faster (we are limiting numbers of re-renders) and it doesn't affect the cursor.